### DISCUSSION TITLE
WGSL ternary operator (?:)

### DISCUSSION THREAD

**Discussion by AshleyScirra** [gpuweb/gpuweb#1391] [Ideas/Feedback]
I had a look through the WGSL spec and couldn't find any references to a ternary operator, e.g. `cnd ? 5 : 7`. Trying it in an actual shader in Chrome Canary produces a syntax error. I also couldn't find any open or closed issues referring to any plans about a ternary operator.

Will WGSL support the ternary operator? Has there been some kind of oversight here? It's useful and a standard feature of many languages - I only found it didn't work when I just went ahead and wrote some WGSL using it and got a syntax error.

**Comment by dj2** [comment]
We have not discussed ternary at this point. I'd probably try to push it to post-MVP at it doesn't provide anything that can't be done in other ways.

**Reply by kvark** [reply]
In addition, if/when we are considering this, we should try just turning `if/else` into an expression, and not introduce a new operator.

**Reply by Kangz** [reply]
Yes, having blocks as expressions would be amazing.
